src/main/java/net/minecraftforge/items/wrapper/VanillaWrapperHandler.java

```diff
@@ -0,0 +1,246 @@
+package net.minecraftforge.items.wrapper;
+
+import net.minecraft.block.Block;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.ISidedInventory;
+import net.minecraft.inventory.InventoryLargeChest;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityChest;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.fml.common.eventhandler.EventPriority;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+
+public class VanillaWrapperHandler {
+
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_CHEST = new ResourceLocation("Forge", "ItemHandler_VanillaWrapperChest");
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_SIDED = new ResourceLocation("Forge", "ItemHandler_VanillaWrapperISided");
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_SIMPLE = new ResourceLocation("Forge", "ItemHandler_VanillaWrapper");
+    private static final ResourceLocation RESOURCE_LOCATION_PASSTHROUGH = new ResourceLocation("Forge", "ItemHandler_PassThrough");
+
+    public static VanillaWrapperHandler INSTANCE = new VanillaWrapperHandler();
+
+    private VanillaWrapperHandler()
+    {
+
+    }
+
+    @SuppressWarnings("unused")
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public void attachVanillaTileEntityWrappers(AttachCapabilitiesEvent.TileEntity event)
+    {
+        TileEntity te = event.getTileEntity();
+        if (te instanceof IItemHandler)
+        {
+            event.addCapability(RESOURCE_LOCATION_PASSTHROUGH, new PassThruWrapper((IItemHandler) te));
+        }
+        else if (te.getClass() == TileEntityChest.class)
+        {
+            event.addCapability(RESOURCE_LOCATION_VANILLA_CHEST, new VanillaWrapperChest((TileEntityChest) te));
+        }
+        else if (te instanceof ISidedInventory)
+        {
+            event.addCapability(RESOURCE_LOCATION_VANILLA_SIDED, new VanillaWrapperSided((ISidedInventory) te));
+        }
```

Comment:
Thats my point I WANT them to implement it themselves so they have that stage of 'Why are pipes not connecting to my stuff? Oh im using a 5 year old tech that I have been to lazy to update!'
You know far well that modders wont update unless they are forced to. And doing the lazy way like this is why I didn't want to add this event in the first place  but fry/cpw bitched.